legend-engine-xts-openapi/legend-engine-xt-openapi-generation/src/test/java/org/finos/legend/engine/generation/TestOpenApiGenerationService.java

```diff
@@ -0,0 +1,48 @@
+//  Copyright 2023 Goldman Sachs
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package org.finos.legend.engine.generation;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.IOUtils;
+import org.finos.legend.engine.generation.model.OpenApiGenerationInput;
+import org.finos.legend.engine.generation.service.OpenApiGenerationService;
+import org.finos.legend.engine.language.pure.modelManager.ModelManager;
+import org.finos.legend.engine.shared.core.ObjectMapperFactory;
+import org.finos.legend.engine.shared.core.deployment.DeploymentMode;
+import org.junit.Test;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+import static org.junit.Assert.assertNotNull;
+
+public class TestOpenApiGenerationService
+{
+    private ObjectMapper objectMapper = new ObjectMapperFactory().getNewStandardObjectMapperWithPureProtocolExtensionSupports();
+
+    @Test
+    public void shouldGenerateOpenApiSpecFromPureModelContext() throws IOException
+    {
+        String payload = IOUtils.toString(Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResource("pure_model_context_payload.json")), StandardCharsets.UTF_8);
+        assertNotNull(payload);
+        OpenApiGenerationInput input = objectMapper.readValue(payload, OpenApiGenerationInput.class);
+        ModelManager manager = new ModelManager(DeploymentMode.TEST);
+        OpenApiGenerationService service = new OpenApiGenerationService(manager);
+        Response response = service.generateOpenApi(input, null);
+        assertNotNull(response);
```

Comment:
we just checking something is generated and not the contents?

Reply:
now asserting the response entity to an expected result

Reply:
do we need assertNull. after checking equality with a string?

Reply:
will clean up the assert null